spec/testsomething_unittests_spec.rb

```diff
@@ -13,7 +13,9 @@ def get_relative_dir(sampleprojects_tests_dir)
 RSpec.describe "TestSomething C++" do
   next if skip_cpp_tests
   cpp_lib_path = sampleproj_path + "TestSomething"
-  cpp_library = ArduinoCI::CppLibrary.new(cpp_lib_path, Pathname.new("my_fake_arduino_lib_dir"))
+  cpp_library = ArduinoCI::CppLibrary.new(cpp_lib_path,
+                                          Pathname.new("my_fake_arduino_lib_dir"),
+                                          ["excludeThis"])
```

Comment:
Please replace `["excludeThis"]` with `[]`, or add tests to the spec file that specifically make refer to the `excludeThis` directory and/or the files it contains.

Reply:
To say this another way: I assume `["excludeThis"]` prevents a regression on the existing tests in this spec file.  That's fine, but you should add a different test section that shows that _without_ excluding a directory, `cpp_library.cpp_files` contains the extra files.

The easiest way I can think of would be to give each `context` section its own `cpp_library = ` line, instead of using a common one.

```ruby
  context "cpp_files without an exclude directory" do
    cpp_library = ArduinoCI::CppLibrary.new(cpp_lib_path, 
                                            Pathname.new("my_fake_arduino_lib_dir"),
                                            [])
    it "finds all cpp files in directory" do
      testsomething_cpp_files = [
        Pathname.new("TestSomething/test-something.cpp"),
        Pathname.new("TestSomething/excludeThis/break-compilation.cpp"),
      ]
      relative_paths = cpp_library.cpp_files.map { |f| get_relative_dir(f) }
      expect(relative_paths).to match_array(testsomething_cpp_files)
    end
  end

  context "cpp_files with an exclude directory" do
    cpp_library = ArduinoCI::CppLibrary.new(cpp_lib_path, 
                                            Pathname.new("my_fake_arduino_lib_dir"),
                                            ["excludeThis"])
    it "finds cpp files in directory and ignores some" do
      testsomething_cpp_files = [Pathname.new("TestSomething/test-something.cpp")]
      relative_paths = cpp_library.cpp_files.map { |f| get_relative_dir(f) }
      expect(relative_paths).to match_array(testsomething_cpp_files)
    end
  end

  context "unit tests" do
    cpp_library = ArduinoCI::CppLibrary.new(cpp_lib_path, 
                                            Pathname.new("my_fake_arduino_lib_dir"),
                                            ["excludeThis"])

```